src/client/Microsoft.Identity.Client/Instance/Discovery/InstanceDiscoveryManager.cs

```diff
@@ -195,5 +217,15 @@ private static InstanceDiscoveryMetadataEntry CreateEntryForSingleAuthority(Uri
                 PreferredNetwork = authority.Host
             };
         }
+
+        private async Task<string> BuildAuthorityWithRegionAsync(string canonicalAuthority)
+        {
+            string region = await _regionDiscoveryProvider.getRegionAsync().ConfigureAwait(false);
+
+            var builder = new UriBuilder(canonicalAuthority);
+            builder.Host = $"{region}.{builder.Host}";
```

Comment:
I believe that we discussed the following rules: 

- if authority is a known public cloud, then regional authority will be `{region}.login.microsoft.com`
- otherwise, the regional authority will be `{region}.{authoirty}`

Reply:
Changed.